vision/automl  unable to update to libraries-bom 5.4.0

Found in #2937
https://source.cloud.google.com/results/invocations/f7079bef-8021-4a51-b88a-24740f041202/targets/cloud-devrel%2Fjava%2Fjava-docs-samples%2Fjava11%2Fpresubmit/log

# Java 11
```
------------------------------------------------------------
- testing vision/automl
------------------------------------------------------------
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 301.586 s <<< FAILURE! - in com.google.cloud.vision.samples.automl.DatasetApiIT
[ERROR] com.google.cloud.vision.samples.automl.DatasetApiIT.testCreateImportDeleteDataset  Time elapsed: 300.524 s  <<< ERROR!
java.util.concurrent.CancellationException: Task was cancelled.
	at com.google.cloud.vision.samples.automl.DatasetApiIT.testCreateImportDeleteDataset(DatasetApiIT.java:73)

[ERROR] Errors:
[ERROR]   DatasetApiIT.testCreateImportDeleteDataset:73 » Cancellation Task was cancelle...
[ERROR] Tests run: 8, Failures: 0, Errors: 1, Skipped: 0
```
@stephaniewang526 PTAL

--- a/vision/automl/pom.xml
+++ b/vision/automl/pom.xml
@@ -43,7 +43,7 @@
       <dependency>
         <groupId>com.google.cloud</groupId>
         <artifactId>libraries-bom</artifactId>
-        <version>4.4.1</version>
+        <version>5.5.0</version>
         <type>pom</type>
         <scope>import</scope>
       </dependency>

--- a/vision/automl/src/main/java/com/google/cloud/vision/samples/automl/DatasetApi.java
+++ b/vision/automl/src/main/java/com/google/cloud/vision/samples/automl/DatasetApi.java
@@ -17,6 +17,7 @@
 package com.google.cloud.vision.samples.automl;
 
 // Imports the Google Cloud client library
+import com.google.api.gax.longrunning.OperationFuture;
 import com.google.cloud.automl.v1beta1.AutoMlClient;
 import com.google.cloud.automl.v1beta1.ClassificationProto.ClassificationType;
 import com.google.cloud.automl.v1beta1.Dataset;
@@ -28,9 +29,12 @@ import com.google.cloud.automl.v1beta1.InputConfig;
 import com.google.cloud.automl.v1beta1.ListDatasetsRequest;
 import com.google.cloud.automl.v1beta1.LocationName;
 import com.google.cloud.automl.v1beta1.OutputConfig;
+import com.google.cloud.automl.v1beta1.OperationMetadata;
 import com.google.protobuf.Empty;
 import java.io.IOException;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import net.sourceforge.argparse4j.ArgumentParsers;
 import net.sourceforge.argparse4j.inf.ArgumentParser;
 import net.sourceforge.argparse4j.inf.ArgumentParserException;
@@ -211,9 +215,21 @@ public class DatasetApi {
       // Import data from the input URI
       InputConfig inputConfig = InputConfig.newBuilder().setGcsSource(gcsSource).build();
       System.out.println("Processing import...");
-      Empty response = client.importDataAsync(datasetFullId.toString(), inputConfig).get();
-      System.out.println(String.format("Dataset imported. %s", response));
-    } catch (IOException | InterruptedException | ExecutionException e) {
+
+      // Start the import job
+      OperationFuture<Empty, OperationMetadata> operation = client
+          .importDataAsync(datasetFullId, inputConfig);
+
+      // More info on gax longrunning Operation:
+      // http://googleapis.github.io/gax-java/1.4.1/apidocs/com/google/api/gax/grpc/OperationFuture.html
+      System.out.format("Operation name: %s%n", operation.getName());
+
+      // If you want to wait for the operation to finish, adjust the timeout appropriately. The
+      // operation will still run if you choose not to wait for it to complete. You can check the
+      // status of your operation using the operation's name.
+      Empty response = operation.get(5, TimeUnit.MINUTES);
+      System.out.format("Dataset imported. %s%n", response);
+    } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
       e.printStackTrace();
     }
   }

--- a/vision/automl/src/main/java/com/google/cloud/vision/samples/automl/DatasetApi.java
+++ b/vision/automl/src/main/java/com/google/cloud/vision/samples/automl/DatasetApi.java
@@ -28,8 +28,8 @@ import com.google.cloud.automl.v1beta1.ImageClassificationDatasetMetadata;
 import com.google.cloud.automl.v1beta1.InputConfig;
 import com.google.cloud.automl.v1beta1.ListDatasetsRequest;
 import com.google.cloud.automl.v1beta1.LocationName;
-import com.google.cloud.automl.v1beta1.OutputConfig;
 import com.google.cloud.automl.v1beta1.OperationMetadata;
+import com.google.cloud.automl.v1beta1.OutputConfig;
 import com.google.protobuf.Empty;
 import java.io.IOException;
 import java.util.concurrent.ExecutionException;
@@ -217,8 +217,8 @@ public class DatasetApi {
       System.out.println("Processing import...");
 
       // Start the import job
-      OperationFuture<Empty, OperationMetadata> operation = client
-          .importDataAsync(datasetFullId, inputConfig);
+      OperationFuture<Empty, OperationMetadata> operation =
+          client.importDataAsync(datasetFullId, inputConfig);
 
       // More info on gax longrunning Operation:
       // http://googleapis.github.io/gax-java/1.4.1/apidocs/com/google/api/gax/grpc/OperationFuture.html

--- a/vision/automl/src/test/java/com/google/cloud/vision/samples/automl/DatasetApiIT.java
+++ b/vision/automl/src/test/java/com/google/cloud/vision/samples/automl/DatasetApiIT.java
@@ -62,7 +62,7 @@ public class DatasetApiIT {
     DatasetApi.createDataset(PROJECT_ID, COMPUTE_REGION, datasetName, false);
 
     // Assert
-    got = bout.toString();
+    String got = bout.toString();
     datasetId =
         bout.toString()
             .split("\n")[0]
@@ -74,7 +74,7 @@ public class DatasetApiIT {
         PROJECT_ID, COMPUTE_REGION, datasetId, "gs://" + BUCKET + "/flower_traindata_concise.csv");
 
     // Assert
-    String got = bout.toString();
+    got = bout.toString();
     assertThat(got).contains("Dataset id:");
 
     // Act

--- a/vision/automl/src/test/java/com/google/cloud/vision/samples/automl/DatasetApiIT.java
+++ b/vision/automl/src/test/java/com/google/cloud/vision/samples/automl/DatasetApiIT.java
@@ -60,29 +60,29 @@ public class DatasetApiIT {
         String.format("test_%s", UUID.randomUUID().toString().replace("-", "_").substring(0, 26));
     // Act
     DatasetApi.createDataset(PROJECT_ID, COMPUTE_REGION, datasetName, false);
-
-    // Assert
-    String got = bout.toString();
+    //
+    // // Assert
+    // String got = bout.toString();
     datasetId =
         bout.toString()
             .split("\n")[0]
             .split("/")[(bout.toString().split("\n")[0]).split("/").length - 1];
-    assertThat(got).contains("Dataset id:");
+    // assertThat(got).contains("Dataset id:");
 
     // Act
     DatasetApi.importData(
-        PROJECT_ID, COMPUTE_REGION, datasetId, "gs://" + BUCKET + "/flower_traindata.csv");
+        PROJECT_ID, COMPUTE_REGION, datasetId, "gs://" + BUCKET + "/flower_traindata_concise.csv");
 
     // Assert
-    got = bout.toString();
+    String got = bout.toString();
     assertThat(got).contains("Dataset id:");
 
-    // Act
-    DatasetApi.deleteDataset(PROJECT_ID, COMPUTE_REGION, datasetId);
-
-    // Assert
-    got = bout.toString();
-    assertThat(got).contains("Dataset deleted.");
+    // // Act
+    // DatasetApi.deleteDataset(PROJECT_ID, COMPUTE_REGION, datasetId);
+    //
+    // // Assert
+    // got = bout.toString();
+    // assertThat(got).contains("Dataset deleted.");
   }
 
   @Test

--- a/vision/automl/src/test/java/com/google/cloud/vision/samples/automl/DatasetApiIT.java
+++ b/vision/automl/src/test/java/com/google/cloud/vision/samples/automl/DatasetApiIT.java
@@ -60,14 +60,14 @@ public class DatasetApiIT {
         String.format("test_%s", UUID.randomUUID().toString().replace("-", "_").substring(0, 26));
     // Act
     DatasetApi.createDataset(PROJECT_ID, COMPUTE_REGION, datasetName, false);
-    //
-    // // Assert
-    // String got = bout.toString();
+
+    // Assert
+    got = bout.toString();
     datasetId =
         bout.toString()
             .split("\n")[0]
             .split("/")[(bout.toString().split("\n")[0]).split("/").length - 1];
-    // assertThat(got).contains("Dataset id:");
+    assertThat(got).contains("Dataset id:");
 
     // Act
     DatasetApi.importData(
@@ -77,12 +77,12 @@ public class DatasetApiIT {
     String got = bout.toString();
     assertThat(got).contains("Dataset id:");
 
-    // // Act
-    // DatasetApi.deleteDataset(PROJECT_ID, COMPUTE_REGION, datasetId);
-    //
-    // // Assert
-    // got = bout.toString();
-    // assertThat(got).contains("Dataset deleted.");
+    // Act
+    DatasetApi.deleteDataset(PROJECT_ID, COMPUTE_REGION, datasetId);
+
+    // Assert
+    got = bout.toString();
+    assertThat(got).contains("Dataset deleted.");
   }
 
   @Test

